lib/internal/async_hooks.js

```diff
@@ -243,27 +246,104 @@ function restoreActiveHooks() {
   active_hooks.tmp_fields = null;
 }
 
+function trackPromise(promise, parent, silent) {
+  const asyncId = getOrSetAsyncId(promise);
+
+  if (parent) {
+    promise[trigger_async_id_symbol] = getOrSetAsyncId(parent);
+    ObjectDefineProperty(promise, 'isChainedPromise', {
```

Comment:
It seems that these `ObjectDefineProperty` calls are quite expensive, at least in the `benchmark/async_hooks/promises.js` scenario. I've tried to remove them locally and got the result comparable with the performance of the initial version (https://github.com/nodejs/node/pull/32891#issuecomment-616055234)

Reply:
The proposed change in included into https://github.com/Qard/node/pull/2

Reply:
Yes, `Object.defineProperty` is significant slower. But it is the only way I know to get a property non enumerable.

Reply:
On the other hand, `Object.defineProperty` seems to be making this optimization meaningless. So, we either have to live with `isChainedPromise` property being enumerable, or remove it completely (but first we need to make sure that nobody is using it).

Reply:
I'm in favor of just removing it, or making it a symbol property instead. It's an obscure corner-case of an already experimental feature. If anyone is using it, they should be _expecting_ that it could disappear at any time. Also, it's already been established that zone.js was the one original user and appears to not be using it anymore. 🤷

Reply:
I think making this enumerable is fine, or alternatively, we could turn it into a symbol property and make `isChainedPromise` a getter on `Promise.prototype` that accesses that symbol?

Reply:
I've merged in the changes from @puzpuzpuz, so the `ObjectDefineProperty` bit is gone for now. I can restore it if we decide we want to go that route, but my suggestion is to just remove the property entirely.

Reply:
I tend to remove. Why keep the burden of an unused feature? If it really turns out that it is needed we can add it again - with some docu why.

Reply:
+1 from me for removing it, as I also failed to find user modules that require this feature. In the worst scenario we can restore it on demand.

Reply:
I added a commit removing `isChainedPromise`. Could also try the symbol property and `Promise.prototype` getter as @addaleax suggested, if we decide we still need it.